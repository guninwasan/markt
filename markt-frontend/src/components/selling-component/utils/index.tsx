export * from "./upload-image";
export * from "./valid-form-details";
