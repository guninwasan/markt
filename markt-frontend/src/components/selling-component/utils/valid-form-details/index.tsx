export * from "./valid-form-details";
