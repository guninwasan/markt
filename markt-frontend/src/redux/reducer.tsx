import { combineReducers } from "@reduxjs/toolkit";
import userAuth from "./slices/user-auth-slice";
import device from "./slices/device-slice";

const reducer = combineReducers({
  userAuth,
  device,
});

export { reducer };
