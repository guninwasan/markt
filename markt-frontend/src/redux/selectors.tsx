import { RootState } from "./store";

const selectors = {
  getUserID: (state: RootState) => state.userAuth.userID,
  getName: (state: RootState) => state.userAuth.name,
  getEmail: (state: RootState) => state.userAuth.email,
  getPhone: (state: RootState) => state.userAuth.phone,
  getUserAuthJWT: (state: RootState) => state.userAuth.jwt,
  getIsLoggedIn: (state: RootState) => state.device.isLoggedIn,
  getIsLoading: (state: RootState) => state.device.isLoading,
  getWishList: (state: RootState) => state.device.wishList,
};

export { selectors };
