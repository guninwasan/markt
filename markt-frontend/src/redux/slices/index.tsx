export * from "./device-slice";
export * from "./user-auth-slice";
