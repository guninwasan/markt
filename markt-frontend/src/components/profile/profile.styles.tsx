import styled from "styled-components";
import { colors } from "../../utils";

const GreetingText = styled.h1`
  font-size: 4rem;
  font-weight: bold;
  color: ${colors.textBlack};
  text-align: left;
  margin-bottom: 1.5rem;
`;

const UserInfo = styled.p`
  font-size: 1.5rem;
  color: ${colors.darkGrey};
  text-align: left;
  margin: 0.5rem 0;
`;

const ProfileDetailsHeading = styled.h2`
  font-size: 2rem;
  font-weight: semi-bold;
  color: ${colors.darkGrey};
  text-align: left;
  margin-bottom: 1rem;
`;

const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;
  margin-top: 2rem;
`;

const Button = styled.button<{ primaryColor?: boolean }>`
  padding: 0.5rem 1rem;
  font-size: 1rem;
  font-weight: bold;
  color: white;
  background-color: ${(props) =>
    props.primaryColor ? colors.darkerPrimary : colors.red};
  border: none;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    background-color: ${(props) =>
      props.primaryColor ? colors.darkerHoverPrimary : colors.redHover};
  }

  &:focus {
    outline: none;
  }
`;

const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
`;

const ModalContent = styled.div`
  background: white;
  padding: 2rem;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
  font-size: 1.2rem;
  text-align: center;
`;

export {
  GreetingText,
  UserInfo,
  ProfileDetailsHeading,
  Button,
  ButtonContainer,
  ModalContent,
  ModalOverlay,
};
