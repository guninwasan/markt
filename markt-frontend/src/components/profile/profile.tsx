import { useState } from "react";
import { useNavigate } from "react-router-dom";
import {
  GreetingText,
  UserInfo,
  ProfileDetailsHeading,
  ButtonContainer,
  Button,
} from "./profile.styles";
import { useSelector } from "react-redux";
import { RootState, selectors, setIsLoading, setIsLoggedIn } from "../../redux";
import { useDispatch } from "react-redux";

const Profile = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const { userName, userEmail, userPhone } = useSelector(
    (state: RootState) => ({
      userName: selectors.getName(state),
      userEmail: selectors.getEmail(state),
      userPhone: selectors.getPhone(state),
    })
  );

  const handleLogout = () => {
    dispatch(setIsLoading(true));
    dispatch(setIsLoggedIn(false));
    setTimeout(() => {
      dispatch(setIsLoading(false));
      navigate("/");
    }, 2000);
  };

  //CHANGE PASSWORD FUNCTIONALITY NEEDS TO BE ADDED
  const handleChangePassword = () => {
    console.log("Changing password");
  };

  return (
    <div>
      <div style={{ padding: "20px", textAlign: "center" }}>
        <GreetingText>Hello, {userName}</GreetingText>
        <ProfileDetailsHeading>Profile Details:</ProfileDetailsHeading>
        <UserInfo>UofT Email Address: {userEmail}</UserInfo>
        {/* <UserInfo>Phone No.: {userPhone}</UserInfo> */}
        <UserInfo>UofT Student ID: {userPhone}</UserInfo>

        <ButtonContainer>
          <Button onClick={handleChangePassword} primaryColor>
            Change Password
          </Button>
          <Button onClick={handleLogout}>Logout</Button>
        </ButtonContainer>
      </div>
    </div>
  );
};

export { Profile };
