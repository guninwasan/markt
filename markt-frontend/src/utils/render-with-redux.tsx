import { render } from "@testing-library/react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux";

export const renderWithRedux = (component: React.ReactNode) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>
  );
};
