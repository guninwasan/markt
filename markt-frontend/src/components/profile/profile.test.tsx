import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import { Profile } from "./profile";
import { renderWithRedux } from "../../utils/render-with-redux";
import { setIsLoading, setIsLoggedIn } from "../../redux";

jest.mock("../../redux", () => ({
  ...jest.requireActual("../../redux"),
  selectors: {
    getName: jest.fn(() => ""),
    getEmail: jest.fn(() => ""),
    getPhone: jest.fn(() => ""),
  },
  setIsLoading: jest.fn(),
  setIsLoggedIn: jest.fn(),
}));

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

describe("Profile", () => {
  beforeEach(() => {
    mockDispatch.mockClear();
  });

  it("should render the profile component with user information", () => {
    renderWithRedux(<Profile />);

    expect(screen.getByText(/Profile Details/i)).toBeInTheDocument();
    expect(screen.getByText(/UofT Email Address:/i)).toBeInTheDocument();
    expect(screen.getByText(/UofT Student ID:/i)).toBeInTheDocument();
  });

  it("should handle logout correctly", () => {
    renderWithRedux(<Profile />);

    const logoutButton = screen.getByText(/Logout/i);
    fireEvent.click(logoutButton);

    expect(mockDispatch).toHaveBeenCalledWith(setIsLoading(true));
    expect(mockDispatch).toHaveBeenCalledWith(setIsLoggedIn(false));
  });

  it("should match the snapshot", () => {
    const { container } = renderWithRedux(<Profile />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
