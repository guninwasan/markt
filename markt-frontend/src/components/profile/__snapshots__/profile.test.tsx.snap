// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot 1`] = `
<div>
  <div
    style="padding: 20px; text-align: center;"
  >
    <h1
      class="sc-blHHSb dZEASA"
    >
      Hello, 
    </h1>
    <h2
      class="sc-egkSDF bUZWBq"
    >
      Profile Details:
    </h2>
    <p
      class="sc-gtLWhw dyUeKS"
    >
      UofT Email Address: 
    </p>
    <p
      class="sc-gtLWhw dyUeKS"
    >
      UofT Student ID: 
    </p>
    <div
      class="sc-fAUdSK kWpjus"
    >
      <button
        class="sc-dntaoT jPTEBr"
      >
        Change Password
      </button>
      <button
        class="sc-dntaoT jxYeMt"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
